src/api/event-services/work-request.js

```diff
@@ -5,12 +5,50 @@ const { cacheGetRequest } = require('../../utils/cache-request');
 const { CacheMissError } = require('../../cache/cache-utils');
 const { handlePagination } = require('../../utils/handlePagination');
 const validateRequest = require('../../utils/schema-validator');
+const getPipelineStatus = require('../general-services/pipeline-status');
+
+const pipelineConstants = require('../general-services/pipeline-manage/constants');
+
+class WorkRequestError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'WorkRequestError';
+  }
+}
+
+const createWorkResponseError = (message) => ({
+  response: {
+    code: 503,
+    error: message,
+  },
+});
 
 
 const handleWorkRequest = async (workRequest, socket) => {
-  const { uuid, pagination } = workRequest;
+  const { uuid, pagination, experimentId } = workRequest;
+
+  // Check if pipeline is runnning
+  const { qc: { status: qcPipelineStatus } } = await getPipelineStatus(
+    experimentId, pipelineConstants.QC_PROCESS_NAME,
+  );
 
   try {
+    if (qcPipelineStatus === pipelineConstants.NOT_CREATED) {
+      throw new WorkRequestError('Work request can not be handled as QC pipeline has not been run');
+    }
+
+    if (qcPipelineStatus === pipelineConstants.RUNNING) {
+      throw new WorkRequestError('Work request can not be handled as a QC pipeline is running');
+    }
+
+    if ([
+      pipelineConstants.ABORTED,
+      pipelineConstants.FAILED,
+      pipelineConstants.TIMED_OUT,
+    ].includes(qcPipelineStatus)) {
+      throw new WorkRequestError('Work request can not be handled because the previous QC pipeline run had an error.');
+    }
+
```

Comment:
I think it is fine to just check if QC pipeline status is not complete, and the actual status to be included in the error message.

Reply:
Ok, simplified to include all status